sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala

```diff
@@ -58,20 +58,19 @@ object ReplaceExpressions extends Rule[LogicalPlan] {
  */
 object ComputeCurrentTime extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = {
-    val currentDates = mutable.Map.empty[String, Literal]
-    val timeExpr = CurrentTimestamp()
-    val timestamp = timeExpr.eval(EmptyRow).asInstanceOf[Long]
-    val currentTime = Literal.create(timestamp, timeExpr.dataType)
+    lazy val currentDate = {
```

Comment:
the overhead of `lazy val` might be bigger than always computing the value. Can we add `lazy val` when we have benchmarks to prove it's necessary?

Reply:
If you have some concerns about performance, I could remove `lazy` or use the approach with `currentDates.getOrElseUpdate`.

Reply:
I'd expect to see
```
...
val currentDate = ...
...
val currentTime = ...
plan transformAllExpressions {
  case CurrentDate() => currentDate
  case CurrentTimestamp() => currentTime
}
```

Reply:
I removed lazy